mars_api/data_import/data_importer.py

```diff
@@ -0,0 +1,143 @@
+    11: {"milestones": int},
+    12: {"awards": int},
+    13: {"greeneries": int},
+    14: {"cities": int},
+    15: {"event_cards": int},
+    16: {"automated_cards": int},
+    17: {"active_cards": int},
+    18: {"resources": int},
+}
+
+
+def import_data(csv_file_path):
+    import csv
+
+    games = []
+    player_scores = []
+    current_game_id = None
+    with open(csv_file_path, newline="") as csvfile:
+        row_reader = csv.reader(csvfile, delimiter=",")
+        print_status("Parsing the csv file")
+
+        for row in row_reader:
+            if current_game_id != row[0]:
+                current_game_id = row[0]
+                games.append(create_game_dict(row))
+            player_scores.append(create_player_score_dict(row, current_game_id))
+
+    print_status(f"Number of games: {len(games)}; player scores: {len(player_scores)}")
+    save_data(games, player_scores)
+    stdout.write(format_message("Import Completed ✅", "\u001b[32;1m"))
+
+
+def save_data(games_dict, player_scores_dict):
+    games_serializer = GameSerializerForImportedData(data=games_dict, many=True)
+    ps_serializer = PlayerScoreSerializer(data=player_scores_dict, many=True)
+    print_status("Validating the data")
+
+    if not games_serializer.is_valid():
+        raise ValueError(format_message(games_serializer.errors))
+
+    print_status("Saving games")
+    try:
+        games_serializer.save()
+    except IntegrityError as err:
+        print_warning(err)
+        del err
+
+    if not ps_serializer.is_valid():
+        raise ValueError(format_message(ps_serializer.errors))
+
+    print_status("Saving scores")
+    try:
+        ps_serializer.save()
+    except IntegrityError as err:
+        print_warning(err)
+        del err
+
+
+def list_to_dict(data, mapping):
+    """Map a list to a dictionary provided the mapping."""
+    dict_ = {}
+    for index, prop_type_map in mapping.items():
+        name_type = list(prop_type_map.items())
+        name = name_type[0][0]
+        type_ = name_type[0][1]
+        # assign the name to the cast data item
+        dict_[name] = type_(data[index])
+
+    return dict_
+
+
+def create_game_dict(data):
+    game_dict = list_to_dict(data, GAME_FIELD_MAPPING)
+
+    if game_dict["game_map"] == FIELD_REMAPPING["game"]["map"]:
+        game_dict["game_map"] = "Tharsis"
+
+    return game_dict
+
+
+def create_player_score_dict(data, game_id):
+    ps_dict = list_to_dict(data, PLAYER_SCORE_FIELD_MAPPING)
+    ps_dict["game"] = game_id
+    ps_dict["player"] = {"nickname": ps_dict["player_nickname"]}
+
+    if ps_dict["corporation"] in FIELD_REMAPPING["player_score"]["corporation"]:
+        ps_dict["corporation"] = FIELD_REMAPPING["player_score"]["corporation"][
+            ps_dict["corporation"]
+        ]
+
+    return ps_dict
+
+
+def print_status(string):
+    stdout.write(f"\u001b[32m{string}\u001b[0m\n")
+
+
+def print_warning(warning):
+    stdout.write(f"\u001b[33mWARNING:\u001b[0m\n{warning}")
```

Comment:
Do you have to use the stdout? What about `logging.Logger`?

Reply:
I just followed the official Django tutorial on writing management commands and they didn't mentioned `logging` class. I've just looked into the source code of how `manage.py` commands are implemented and they all use `self.stdout.write`. 
What would be the advantage of using `logging`?

Reply:
https://stackoverflow.com/questions/14058453/making-python-loggers-output-all-messages-to-stdout-in-addition-to-log-file look at the way it gets configured and how it's used - it's the standard way of logging in python

Reply:
yes, I've used `colorlog`, just about to push it